src/crawlee/abstract_http_crawler/_abstract_http_crawler.py

```diff
@@ -204,13 +204,10 @@ async def _handle_blocked_request(
         """
         if self._retry_on_blocked:
             status_code = context.http_response.status_code
-
-            # TODO: refactor to avoid private member access
-            # https://github.com/apify/crawlee-python/issues/708
-            if (
-                context.session
-                and status_code not in self._http_client._ignore_http_error_status_codes  # noqa: SLF001
-                and context.session.is_blocked_status_code(status_code=status_code)
+            if context.session and context.session.is_blocked_status_code(
```

Comment:
Did you check the equivalent method in `PlaywrightCrawler` as well? Any chance we could deduplicate this?

Reply:
This is possible after refactoring as @Pijukatel did with `AbstractHttpParser`. So for now, I've only added a general method for dealing with status codes.

Reply:
Nice, thank you!